src/utils.rs

```diff
@@ -50,3 +54,42 @@ pub fn parse_gj_messages_response(meow: String) -> Result<Vec<GDMessage>, Generi
         })
         .collect()
 }
+
+pub async fn checkdiscordusername(mut conn: Connection<Db>, discord_snowflake: String, id: u32) {
+    let time = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs();
+    sqlx::query_as!(DBUser, "UPDATE users SET check_timeout = ? WHERE id = ?", time, id).execute(&mut **conn).await.unwrap();
+
+    let client = Client::new();
+
+    let regex = Regex::new(r".*:").unwrap();
+    let snowflake = regex.replace_all(&discord_snowflake, "");
+
+    let regex = Regex::new(r":|\d").unwrap();
+    let username = regex.replace_all(&discord_snowflake, "");
+
+    //dbg!(username);
+
+    let req = client.get(format!("https://discord.com/api/v9/users/{}", snowflake))
+    .header("Authorization", format!("Bot {}", dotenvy::var("DC_BOT_TOKEN").unwrap()))
+    .send()
+    .await
+    .unwrap()
+    .text()
+    .await
+    .unwrap();
+
+    let crack = ajson::get(&req, "username").unwrap().unwrap();
```

Comment:
- don't use unwrap unless its *really* necessary - user proper error handling
- as mentioned above, please don't add another pointless dependency - use whats already there (serde_json)
- name this something like "real_username" (not crack)